packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -452,6 +452,12 @@ private void handleImageResult(String path) {
       Double maxHeight = methodCall.argument("maxHeight");
 
       String finalImagePath = imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight);
+
+      //delete original file if scaled
+      if (!finalImagePath.equals(path)) {
```

Comment:
Maybe we can put this block of code after `finishWithSuccess(finalImagePath);`? So the dart side can get the result a little earlier?